packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp

```diff
@@ -91,12 +91,12 @@ void Trilinos_Util_ReadHpc2Epetra_internal(
   // the warning is easy to fix.
   if(sizeof(int) == sizeof(int_type)) {
     int numGlobalEquations_int, total_nnz_int;
-#ifdef DEBUG
+#ifndef NDEBUG
     cnt =
 #endif
     fscanf(in_file,"%d",&numGlobalEquations_int);
     assert(cnt > 0);
```

Comment:
Doesn't this line need to be inside one of the `#if` blocks?

Reply:
@jmgate It's fine -- `assert` has the following definition, according to https://en.cppreference.com/w/cpp/error/assert :
```c++
#ifdef NDEBUG
#define assert(condition) ((void)0)
#else
#define assert(condition) /*implementation defined*/
#endif
```

Reply:
The macro `assert()` is only defined if the macro `NDEBUG` is **not** defined.

Reply:
Does the fact that all other occurrences of `cnt` are wrapped in an `#if` not come into play here?  @mhoemmen, it sounds like you're saying line 98 gets replaced with `((void)0)` so it won't throw this error again?
```
error: ‘cnt’ was not declared in this scope
```

I'm asking Santa for a Trilinos without `#ifdef`s this year.

Reply:
@jmgate wrote:
> @mhoemmen, it sounds like you're saying line 98 gets replaced with `((void)0)` so it won't throw this error again?

Yes, that's correct.  `assert` is a macro.  If `NDEBUG` is defined, then the macro's argument does not appear in the compiled source code.  That is, in fact, why the compiler was warning about "unused variables."

> I'm asking Santa for a Trilinos without `#ifdef`s this year.

I ... don't have anything constructive to say about this ;-)